src/AdminAcceptanceTests.Actions/Utils/EmailServerDriver.cs

```diff
@@ -25,23 +24,33 @@ public static async Task<IEnumerable<Email>> FindAllEmailsAsync(string hostUrl,
         {
             using var client = NewHttpClient();
             var response = await client.GetAsync(GetAllEmailsUrl(hostUrl));
-            var responseContent = JToken.Parse(await response.Content.ReadAsStringAsync());
-
-            var emailList = responseContent.Select(x => new Email
+            if (response.IsSuccessStatusCode)
             {
-                Id = x.SelectToken("id").ToString().Trim(),
-                PlainTextBody = x.SelectToken("text").ToString().Trim(),
-                HtmlBody = x.SelectToken("html").ToString().Trim(),
-                Subject = x.SelectToken("subject").ToString(),
-                From = x.SelectToken("from").First().SelectToken("address").ToString(),
-                To = x.SelectToken("to").First().SelectToken("address").ToString(),
-            });
+                var content = await response.Content.ReadAsStringAsync();
 
-            if (emailToCheck != null)
-            {
-                emailList = emailList.Where(e => e.To.Equals(emailToCheck, StringComparison.OrdinalIgnoreCase));
+                // Filters out the emails that are not relevant for the admin tests
+                var responseContent = JToken.Parse(content)
+                    .Where(s => s.SelectToken("subject").ToString().Contains("password", StringComparison.OrdinalIgnoreCase));
+
+                var emailList = responseContent
+                    .Select(x => new Email
+                    {
+                        Id = x.SelectToken("id").ToString().Trim(),
+                        PlainTextBody = x.SelectToken("text").ToString().Trim(),
+                        HtmlBody = x.SelectToken("html").ToString().Trim(),
+                        Subject = x.SelectToken("subject").ToString(),
+                        From = x.SelectToken("from").First().SelectToken("address").ToString(),
+                        To = x.SelectToken("to").First().SelectToken("address").ToString(),
+                    });
+
+                if (emailToCheck != null)
+                {
+                    emailList = emailList.Where(e => e.To.Equals(emailToCheck, StringComparison.OrdinalIgnoreCase));
+                }
+                return emailList;
             }
-            return emailList;
+
+            throw new Exception();
```

Comment:
Is there a more specific exception type you can throw? Would `InvalidOperationException` fit? Is there a message you could include that might help explain the possible cause of the exception if someone less familiar with the tests than you was working on/running them?